service/src/main/java/zone/cogni/asquare/service/async/AsyncUtils.java

```diff
@@ -1,6 +1,5 @@
 package zone.cogni.asquare.service.async;
 
-import javassist.bytecode.annotation.NoSuchClassError;
```

Comment:
@alexey-lukashov we must also fix build.gradle

Reply:
this dependency could be excluded from service project but it will stay in A2 anyway bcos **service projet** imports **application-profile** imports **spring-boot-starter-thymeleaf** imports **javassist**

Reply:
spring-boot-starter-thymeleaf is used for TemplateFactory from application-profile

Reply:
yes, true. but with the fix we do not need to explicitly add it gradle anymore.
so, let's remove it. after removing it, dependencies will look like

```
|    |    +--- org.springframework.boot:spring-boot-starter-thymeleaf -> 2.2.9.RELEASE
|    |    |    +--- org.springframework.boot:spring-boot-starter:2.2.9.RELEASE (*)
|    |    |    +--- org.thymeleaf:thymeleaf-spring5:3.0.11.RELEASE
|    |    |    |    +--- org.thymeleaf:thymeleaf:3.0.11.RELEASE
|    |    |    |    |    +--- ognl:ognl:3.1.12
|    |    |    |    |    |    \--- org.javassist:javassist:3.20.0-GA
```

where before it required to be explicit, one of the "root" dependencies

```
+--- org.javassist:javassist -> 3.20.0-GA
+--- org.junit.jupiter:junit-jupiter -> 5.5.2
```

Reply:
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 8b4aed3a19d3e42e801ab9c63fe27c31fd8275be)
+++ b/build.gradle	(date 1633126824240)
@@ -95,7 +95,6 @@
       // netflix zuul 2.2.3 is aligned with spring boot 2.2.9!
       dependency("org.springframework.cloud:spring-cloud-starter-netflix-zuul:2.2.3.RELEASE")
       dependency("ognl:ognl:3.1.12") // seems thymeleaf needs this library
-      dependency('org.javassist:javassist:3.20.0-GA')
 
       // test
       dependency('org.assertj:assertj-core:3.16.1')
@@ -455,7 +454,6 @@
     // spring
     implementation('org.springframework.boot:spring-boot-starter-aop')
     implementation('org.springframework.boot:spring-boot-starter-json')
-    implementation('org.javassist:javassist')
 
     // test
     testImplementation('org.junit.jupiter:junit-jupiter')

Reply:
```
Index: build.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/build.gradle b/build.gradle
--- a/build.gradle	(revision 8b4aed3a19d3e42e801ab9c63fe27c31fd8275be)
+++ b/build.gradle	(date 1633126824240)
@@ -95,7 +95,6 @@
       // netflix zuul 2.2.3 is aligned with spring boot 2.2.9!
       dependency("org.springframework.cloud:spring-cloud-starter-netflix-zuul:2.2.3.RELEASE")
       dependency("ognl:ognl:3.1.12") // seems thymeleaf needs this library
-      dependency('org.javassist:javassist:3.20.0-GA')
 
       // test
       dependency('org.assertj:assertj-core:3.16.1')
@@ -455,7 +454,6 @@
     // spring
     implementation('org.springframework.boot:spring-boot-starter-aop')
     implementation('org.springframework.boot:spring-boot-starter-json')
-    implementation('org.javassist:javassist')
 
     // test
     testImplementation('org.junit.jupiter:junit-jupiter')

```

Reply:
ok fixed, I havent noticed this was included in build.gradle, there is no such deendency in bitbucket version

Reply:
Indeed. It seems Gradle 7 behaves a bit differently compared to 4 or 5.